### ISSUE TITLE
Examine the assimilation of radiosonde observations - GSI vs JEDI for the MPAS case

### ISSUE THREAD

**Issue by guoqing-noaa** [NOAA-EMC/RDASApp#83]
We will examine the assimilation of radiosonde observations - GSI vs JEDI for the MPAS case
The general plan is to do a single observation test using the MPAS test case included in RDASApp.
Then we will do a full test of radiosonde DA.

The reason to create an issue separate from the FV3 case is that for the MPAS case, we will run the MPASSIT progrm to convert the MPAS grid to a HRRR-like lambert grid and we can conduct similar DA tests based on that grid and compare results to MPAS-JEDI.

@Junjun-NOAA  is working on running MPASSIT.

**Comment by Junjun-NOAA** [comment]
The plan is to transform RDASApp MPAS test data (both deterministic and ensemble) into HRRR-like lambert grid.  When using MPASSIT, we need to defined three files: a grid file to provide the grid information, a history MPAS data file and a diag MPAS file for the diagnostic variables. In RDASApp, we can only find one file. Using the deterministic case as example, we can find the background file from rrfs-test-data/rrfs-data_mpasjedi_2022052619/Data/bkg/bg.2022-05-26_19.00.00.nc. I tried to set all these three files to be the bg.2022-05-26_19.00.00.nc, the program crashed due to some missing values in the grid file (e.g. ter field). I have some questions to ask:

1) I am supposing the bg file is from MPAS forecasts. Can you please provide the initial file and the forecast files? Are the history and diag files the same file? If not, can you provide these three(or two) files you were using when creating the bg file.

2) MPASIT also defined the variables to be transferred from MPAS data to HRRR-like grid. I found there are several variables GSL is using are missed in this bg file. Can you please provide the variable list you included when creating the bg file? 

@guoqing-noaa @TingLei-NOAA

**Comment by guoqing-noaa** [comment]
Thank @Junjun-NOAA for working on this.
I am not sure who created the MPASJEDI case in the current RDASApp. So I would like to loop more people in this thread:
@ShunLiu-NOAA @TingLei-NOAA @xyzemc @SamuelDegelia-NOAA @delippi 
Could anyone help answer @Junjun-NOAA's question in the above? Thanks!

This issue is also related to the issue: https://github.com/NOAA-EMC/RDASApp/issues/84
If we prepare a 30-member MPASJEDI case, how do you get the bkg file? Is it the mpasout file?

**Comment by TingLei-NOAA** [comment]
@Junjun-NOAA 1) In our current mpas-jedi test from @xyzemc 's 15km mpas forecast,, we use restart files for background and ensembles. For example, restart.2022-05-26_19.00.00.nc was copied to  bg.2022-05-26_19.00.00.nc.  The dir containing the forecast from Xiaoyan is hera: /scratch2/NCEPDEV/fv3-cam/Ting.Lei/dr-mpas-bundle/rdas-mpas-tests/CONUS_15km_forecast, you could find the diag files too.
 2) for how to specify the variables in output, i think @xyzemc  is the better person to answer you.  From my side, they were controlled by those stream files in the same directory : like those files defined by  <file name> key in streams.atmosphere . 
 But in this case, the <file name> for restart files is missing and I assume the default setup was used.   
Again, @xyzemc  could clarify on this. 
Hope this could help before Xiaoyan clarify on remaining confusion:).

**Comment by guoqing-noaa** [comment]
@TingLei-NOAA Thanks a lot for the information!

I have a question on this file:
`/scratch2/NCEPDEV/fv3-cam/Ting.Lei/dr-mpas-bundle/rdas-mpas-tests/CONUS_15km_forecast/streams.atmosphere
`
```
<stream name="output"
        type="output"
        filename_template="mpasout.$Y-$M-$D_$h.$m.$s.nc"
        output_interval="1:00:00" >

        <file name="stream_list.atmosphere.output"/>
</stream>

```

Did you modify 
https://github.com/MPAS-Dev/MPAS-Model/blob/develop/src/core_atmosphere/Registry.xml
so that the above "output" stream can output variables listed in the `da_state` stream which the mpasout files should contain?

**Comment by TingLei-NOAA** [comment]
@guoqing-noaa  This part definitely needs more investigation. 
Currently, 1) what I do is to use the same background file to be renamed with the analysis file like an.**  before analysis begins. I think this is the mpas-jedi's standard way. 
2). What I understand is that the updated variables in the pre-existing analysis file would be the analysis variables (directly or indirectly) which is controlled by mpas-jedi and assume the  "stream_list.atmosphere.output" will not affect this process.      
Basically, I think this part is similar to GSI fv3lam 's procedure. 
Further verification on this will be helpful.

**Comment by xyzemc** [comment]
> @TingLei-NOAA Thanks a lot for the information!
> 
> I have a question on this file: `/scratch2/NCEPDEV/fv3-cam/Ting.Lei/dr-mpas-bundle/rdas-mpas-tests/CONUS_15km_forecast/streams.atmosphere `
> 
> ```
> <stream name="output"
>         type="output"
>         filename_template="mpasout.$Y-$M-$D_$h.$m.$s.nc"
>         output_interval="1:00:00" >
> 
>         <file name="stream_list.atmosphere.output"/>
> </stream>
> ```
> 
> Did you modify https://github.com/MPAS-Dev/MPAS-Model/blob/develop/src/core_atmosphere/Registry.xml so that the above "output" stream can output variables listed in the `da_state` stream which the mpasout files should contain?

I didn't modify the Registry.xml when I created the case. So just the default variables are included in the mpasout file. At the beginning, we tried to use the mpasout as the analysis background, but some variables are missing. So we switch to use the restart file. I think if you modify the Registry.xml to add are required variables, the mpasout file should work for the JEDI background.

**Comment by guoqing-noaa** [comment]
@xyzemc Thanks for the clarification.  @TingLei-NOAA Thanks for the information.
Now I got it. The mpasout files in your directory are actually history files.

In order to generate mpasout files, we will need to use the MPASJEDI version of the MPAS model. I checked with NCAR MPASJEDI developers and the good news is that we are not far from this MPASOUT function being integrated into the latest MPAS model version.

**Comment by guoqing-noaa** [comment]
I agree that we may use the restart files at the moment.

**Comment by guoqing-noaa** [comment]
@xyzemc 

in streams.atmosphere_15km, there is a templateFields.93175.nc file:
```
immutable_stream name="input"                                                                                                                                         
                  type="input"
                  precision="single"
                  filename_template="Data_static/templateFields.93175.nc"
                  io_type="pnetcdf,cdf5"
                  input_interval="initial_only" />
```
Is this required by MPAS JEDI?  How did you generate this file? Thanks!

Is this a renaming of the` init.nc` file?

**Comment by Junjun-NOAA** [comment]
Thanks @xyzemc and @TingLei-NOAA for the information. The below is what I have got from mpassit (T2 field), can you please check whether the domain is correct? Thanks.
<img width="452" alt="Screenshot 2024-06-20 at 8 11 22 PM" src="https://github.com/NOAA-EMC/RDASApp/assets/151088579/6a069349-1f17-4185-bbf4-9d052943b025">

**Comment by xyzemc** [comment]
> Thanks @xyzemc and @TingLei-NOAA for the information. The below is what I have got from mpassit (T2 field), can you please check whether the domain is correct? Thanks. <img alt="Screenshot 2024-06-20 at 8 11 22 PM" width="452" src="https://private-user-images.githubusercontent.com/151088579/341606022-6a069349-1f17-4185-bbf4-9d052943b025.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg5NzI3MjEsIm5iZiI6MTcxODk3MjQyMSwicGF0aCI6Ii8xNTEwODg1NzkvMzQxNjA2MDIyLTZhMDY5MzQ5LTFmMTctNDE4NS1iYmY0LTlkMDUyOTQzYjAyNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyMVQxMjIwMjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zYzRhZGY5ZmEzMmJiNTRhM2I3ZTIwYjc5YTY3N2U3ZGM0NDMwYzAzNjJhNzVjYTE4NTRmMDViMzUxM2ZhMDFjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Y2uKcFq849qnygxaeafMseZ3SQS8awRHhFGTjOEvzbk">

Yes, the domain looks right. You can check this [link ](https://private-user-images.githubusercontent.com/45010998/327883367-9399e2e3-8ca5-49dd-a7fb-93d8ae9491b3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg5NzI1NDYsIm5iZiI6MTcxODk3MjI0NiwicGF0aCI6Ii80NTAxMDk5OC8zMjc4ODMzNjctOTM5OWUyZTMtOGNhNS00OWRkLWE3ZmItOTNkOGFlOTQ5MWIzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjIxVDEyMTcyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNlMGViN2FiZDU4NzMzZmJkYTE0OTczN2JkNjgzZjQ4ZDc5M2M1ZTRhNWJhYTAzZjU1NGYyNTI5MTM5OTUyODUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.z50_m6281QWkVbT1TgBLIDDxHMeyuldNmDZrHKhimtA)to see the domain plot that I created on the global map.

**Comment by xyzemc** [comment]
> The plan is to transform RDASApp MPAS test data (both deterministic and ensemble) into HRRR-like lambert grid. When using MPASSIT, we need to defined three files: a grid file to provide the grid information, a history MPAS data file and a diag MPAS file for the diagnostic variables. In RDASApp, we can only find one file. Using the deterministic case as example, we can find the background file from rrfs-test-data/rrfs-data_mpasjedi_2022052619/Data/bkg/bg.2022-05-26_19.00.00.nc. I tried to set all these three files to be the bg.2022-05-26_19.00.00.nc, the program crashed due to some missing values in the grid file (e.g. ter field). I have some questions to ask:
> 
> 1. I am supposing the bg file is from MPAS forecasts. Can you please provide the initial file and the forecast files? Are the history and diag files the same file? If not, can you provide these three(or two) files you were using when creating the bg file.
> 2. MPASIT also defined the variables to be transferred from MPAS data to HRRR-like grid. I found there are several variables GSL is using are missed in this bg file. Can you please provide the variable list you included when creating the bg file?
> 
> @guoqing-noaa @TingLei-NOAA

If you are still looking for the diagnostic file, you can find it from my directory on Hera: /scratch1/NCEPDEV/fv3-cam/Xiaoyan.Zhang/mpas/regional/CONUS_15km/JEDI_MPAS_CONUS/diag

**Comment by xyzemc** [comment]
> @xyzemc
> 
> in streams.atmosphere_15km, there is a templateFields.93175.nc file:
> 
> ```
> immutable_stream name="input"                                                                                                                                         
>                   type="input"
>                   precision="single"
>                   filename_template="Data_static/templateFields.93175.nc"
>                   io_type="pnetcdf,cdf5"
>                   input_interval="initial_only" />
> ```
> 
> Is this required by MPAS JEDI? How did you generate this file? Thanks!
> 
> Is this a renaming of the` init.nc` file?

 The Template file was handled by @TingLei-NOAA . It looks like copied from init.nc, but let's wait for  @TingLei-NOAA to confirm it.

**Comment by guoqing-noaa** [comment]
> The Template file was handled by @TingLei-NOAA . It looks like copied from init.nc, but let's wait for @TingLei-NOAA to confirm it.

@xyzemc, thanks for the information! I used a restart file to replace this templateFields file, it looks like it worked for this step.

**Comment by guoqing-noaa** [comment]
Increments from a single sonde ob 3DEnVAR experiment with 401km horizontal JEDI radius and 3 vertical levels radius.
It looks like the ensembles don't provide much flow-dependent structure for this case. 

![image](https://github.com/NOAA-EMC/RDASApp/assets/51679803/216d46d7-18ce-48cd-a51f-b6f52135e63b)

**Comment by guoqing-noaa** [comment]
Increments from a single sonde ob 3DEnVAR experiment with 401km horizontal JEDI radius and 11 vertical levels radius (this is equivalent to 3 e-folding radius in GSI)

![image](https://github.com/NOAA-EMC/RDASApp/assets/51679803/1f2fb2db-8ec4-4d3a-bfa6-a38451d83af1)
